src/Future.js

```diff
@@ -10,6 +10,16 @@ function jail(handler, f){
   };
 }
 
+function fork(m, reject, resolve) {
```

Comment:
As mentioned in my other comment, I wouldn't be concerned with trying to make this play nicely with other things that look like a `Future` for now. If we were to go down that path, we would first need to spend some time defining a spec around expected behaviour.